src/core/net/udp6.cpp

```diff
@@ -419,9 +417,9 @@ uint16_t Udp::GetEphemeralPort(void)
         {
             mEphemeralPort = kDynamicPortMin;
         }
-    } while (rval == Tmf::kUdpPort);
+    } while (mEphemeralPort == Tmf::kUdpPort);
```

Comment:
```suggestion
    } while (IsPortInUse(mEphemeralPort));
```

Reply:
What about `IsPortReserved`? I guess it's not easy to implement a function returning ports being currently used.

Reply:
I think both ports needs to be skipped:
1) Ports reserved: TMF port only
2) Ports in use

But this is out of the scope of this PR. I think this PR should focus on fixing this apparent bug. 
We can create another PR to enhance 2) later.

Thoughts? @wgtdkp @bukepo @superwhd

Reply:
I agree to focus on fixing the bug in this PR.